### PR TITLE
Task 17/implement ESLint with revert fix

### PR DESCRIPTION
**Pull request by JimLau49**
This issue references issue #52,

ESLint is now implemented in our project and is deployed on Travis. 

To run(Updated on README.md):
`npm run lint` : to run the linter
`npm run lint-fix` : allow ESLint to attempt to fix the issues

Note: Only .eslintrc.js, travis.yml, tsconfig.json, package.json and README.md are the only files that were changed by us, the rest were all changes done by the linter.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:react/recommended"
+  ],
+  parserOptions: {
+    project: "./tsconfig.json",
+    tsconfigRootDir: __dirname
+  },
+  rules: {
+    "@typescript-eslint/no-use-before-define": [
+      "error",
```
Comment (RGPosadas) [inline .eslintrc.js:19]: This plugin has 4 out of 5 checks, but we are silencing 4. What is the use of this plugin then? Unless you set some to true

**Review by RGPosadas — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (JimLau49) [reply .eslintrc.js:19]: This is actually really true, we're gonna fix this up so thats it only for variables so only 1/5 is silenced

File: package.json [hunk 2]
```diff
@@ -7,9 +7,12 @@
     "ios": "expo start --ios",
     "start": "expo start",
     "test": "jest --verbose --coverage",
-    "web": "expo start --web"
+    "web": "expo start --web",
+    "lint": "tsc --noEmit && eslint --ext .ts,.tsx ./",
+    "lint-fix": "tsc --noEmit && eslint --fix --ext .ts,.tsx ./"
   },
   "dependencies": {
+    "@expo/vector-icons": "^10.0.6",
```
Comment (cristian-aldea) [inline package.json:15]: I believe that this is not necessary. this package in included in `expo`
https://docs.expo.io/versions/latest/guides/icons/

**Review by RGPosadas — APPROVED** [review]
Ran the app, all good on Android (physical Pixel 3) + iOS (iPhone 11 Max Pro emulator).
LGTM!

**Review by cristian-aldea — APPROVED** [review]
Looks good. Left a comment that should be addressed ideally. Won't be blocking however

File: package.json (hunk 2 above)
Comment (leojrsilao) [reply package.json:15]: You're right it passes locally without the package dependency, but it fails on Travis which is why we need it.

**Pull request merged**